repository/src/main/java/org/alfresco/repo/rendition2/RenditionService2Impl.java

```diff
@@ -598,6 +570,184 @@ else if (!nodeService.hasAspect(renditionNode, RenditionModel.ASPECT_RENDITION2)
         }
     }
 
+    /**
```

Comment:
Can this be moved to switchable RenditionService3Impl (potentially extends RenditionService2Impl, if that is easier initially) ?